lobster/tools/codebeamer/codebeamer.py

```diff
@@ -373,108 +378,102 @@ def ensure_array_of_strings(instance):
         return [str(instance)]
 
 
-def parse_cb_config(file_name):
-    assert isinstance(file_name, str)
-    assert os.path.isfile(file_name)
+def parse_yaml_config(file_name: str):
+    """
+    Parses a YAML configuration file and returns a validated configuration dictionary.
 
-    with open(file_name, "r", encoding='utf-8') as file:
-        data = json.loads(file.read())
+    Args:
+        file_name (str): Path to the YAML configuration file.
 
-    json_config = {REFERENCES: {}}
+    Returns:
+        Dict[str, Any]: Parsed and validated configuration.
 
-    if TOKEN in data:
-        json_config["token"] = data.pop(TOKEN)
+    Raises:
+        ValueError: If `file_name` is not a string.
+        FileNotFoundError: If the file does not exist.
+        KeyError: If required fields are missing or unsupported keys are present.
+    """
+    if not isinstance(file_name, str):
+        raise ValueError("File name must be a string.")
+    if not os.path.isfile(file_name):
+        raise FileNotFoundError(f"File not found: {file_name}")
+
+    default_values = {
+        'timeout': 30,
+        'page_size': 100,
+        'verify_ssl': False,
+        'schema': 'Requirement',
+    }
 
-    provided_config_keys = set(data.keys())
-    schema = data.get("schema", "Requirement").lower()
+    required_fields = {"import_tagged", "import_query"}
 
-    if not provided_config_keys.issubset(SupportedConfigKeys.as_set()):
-        raise KeyError("The provided config keys are not supported! "
-                       "supported keys: '%s'" %
-                       ', '.join(SupportedConfigKeys.as_set()))
+    with open(file_name, "r", encoding='utf-8') as file:
+        data = yaml.safe_load(file) or {}
+
+    # Ensure at least one required field is present
+    if not required_fields & data.keys():
+        raise KeyError(f"One of the required fields "
+                       f"must be present: {', '.join(required_fields)}")
+
+    # Build the configuration dictionary
+    json_config = {
+        "references": {
+            "refs": ensure_array_of_strings(data["refs"])
+        } if "refs" in data else {},
+        "token": data.pop("token", None),
+        "base": f"{data.get('root', '')}/cb/api/v3",
+    }
 
-    for key, value in data.items():
-        json_config[REFERENCES][key] = ensure_array_of_strings(value)
+    # Validate supported keys
+    provided_config_keys = set(data.keys())
+    unsupported_keys = provided_config_keys - SupportedConfigKeys.as_set()
+    if unsupported_keys:
+        raise KeyError(
+            f"Unsupported config keys: {', '.join(unsupported_keys)}. "
+            f"Supported keys are: {', '.join(SupportedConfigKeys.as_set())}."
+        )
```

Comment:
Same as https://github.com/bmw-software-engineering/lobster/pull/137/files#r1875918025

Reply:
As we discussed, each tool has different parameters, so it can't be moved into utils.